api/src/routes/email-subscription.ts

```diff
@@ -0,0 +1,136 @@
+ */
+export const emailSubscribtionRoutes: FastifyPluginCallbackTypebox = (
+  fastify,
+  _options,
+  done
+) => {
+  fastify.get(
+    '/ue/:unsubscribeId',
+    {
+      schema: schemas.unsubscribe,
+      errorHandler(error, request, reply) {
+        if (error.validation) {
+          const { origin } = getRedirectParams(request);
+          void reply.code(302);
+          void reply.redirectWithMessage(origin, {
+            type: 'info',
+            content: 'We could not find an account to unsubscribe.'
+          });
+        } else {
+          fastify.errorHandler(error, request, reply);
+        }
+      }
+    },
+    async (req, reply) => {
+      try {
+        const { origin } = getRedirectParams(req);
+        const { unsubscribeId } = req.params;
+        const users = await fastify.prisma.user.findMany({
+          where: { unsubscribeId }
+        });
+
+        if (!users.length) {
+          void reply.code(302);
+          return reply.redirectWithMessage(origin, {
+            type: 'info',
+            content: 'We could not find an account to unsubscribe.'
+          });
+        }
+
+        for (const user of users) {
+          await fastify.prisma.user.update({
+            where: { id: user.id },
+            data: {
+              sendQuincyEmail: false
+            }
+          });
+        }
+
+        return reply.redirectWithMessage(
+          `${origin}/unsubscribed/${unsubscribeId}`,
+          {
+            type: 'success',
+            content: "We've successfully updated your email preferences."
+          }
+        );
+      } catch (error) {
+        fastify.log.error(error);
+        void reply.code(302);
+        return reply.redirectWithMessage(origin, {
+          type: 'danger',
+          content: 'Something went wrong.'
+        });
+      }
+    }
+  );
+
+  fastify.get(
+    '/resubscribe/:unsubscribeId',
+    {
+      schema: schemas.resubscribe,
+      errorHandler(error, request, reply) {
+        if (error.validation) {
+          const { origin } = getRedirectParams(request);
+          void reply.code(302);
+          void reply.redirectWithMessage(origin, {
+            type: 'info',
+            content:
+              'We were unable to process this request, please check and try again.'
+          });
+        } else {
+          fastify.errorHandler(error, request, reply);
+        }
+      }
+    },
+    async (req, reply) => {
+      try {
+        const { origin } = getRedirectParams(req);
+        const { unsubscribeId } = req.params;
+        const users = await fastify.prisma.user.findMany({
+          where: { unsubscribeId }
+        });
+
+        if (!users.length) {
+          void reply.code(302);
+          return reply.redirectWithMessage(origin, {
+            type: 'info',
+            content: 'We could not find an account to resubscribe.'
+          });
+        }
+
+        await fastify.prisma.user.update({
+          where: { id: users[0]?.id },
```

Comment:
This just makes the type better.

```suggestion
        const [user] = await fastify.prisma.user.findMany({
          where: { unsubscribeId }
        });

        if (!user) {
          void reply.code(302);
          return reply.redirectWithMessage(origin, {
            type: 'info',
            content: 'We could not find an account to resubscribe.'
          });
        }

        await fastify.prisma.user.update({
          where: { id: user.id },
```

Reply:
Why are we using `findMany` here if we're only looking at the first one?

Reply:
Because we are rewriting the old api - bugs included.

Reply:
This makes me cry

Reply:
No, this would introduce a new bug.

`unsubscribeId` is not unique as far as the db is concerned, it just knows it's a string, so we cannot use `findUnique`. We could use `findFirst`, but we should just use `findMany` and unsubscribe them all.

There's roughly 0 chance of this being necessary, but it guards against us messing up and using the same `unsubscribeId` twice.

Reply:
But that doesn't seem like what we're doing?

Reply:
Right. We should do what the old api did. The implementation was fine.

Reply:
so this is fine as-is?

Reply:
This is fine. Either implementation works. My suggestion just makes TypeScript happier.